### DISCUSSION TITLE
Validation not working with parent component and child component

### DISCUSSION THREAD

**Discussion by kevinkli916** [react-hook-form/react-hook-form#1771] [General] [answered]
Hello,

I am having trouble getting validation to work when the parent component contains the <form> wrapping the child component containing all other RHF logic.

Here is a codesandbox [link](https://codesandbox.io/s/react-hook-form-errormessage-gm9d6?file=/src/index.js)

Any help is appreciated.

**Comment by bluebill1049** [comment, marked as answer]
please explain your question a bit clearer and also what's the steps to reproduce the problem. comments inside codesandbox don't really help much.

**Reply by kevinkli916** [reply]
Hi Bill,

So in [this CSB](https://codesandbox.io/s/react-hook-form-errormessage-gm9d6?file=/src/index.js), the submit button is not properly validating the components. However, when I move the contents of SelectComponent into index.js, validation beings to work again... such as in [this other CSB](https://codesandbox.io/s/intelligent-christian-nl8ux?file=/src/index.js). I suspect it has to do with the parent component containing the form tags wrapping the child component <SelectComponent/>, but I do not know how to fix it.

**Reply by bluebill1049** [reply]
can you please start with this page https://react-hook-form.com/get-started? I think you missing some fundamentals of how this custom hook works. you have two custom hooks inside your example, and the submit button is at the root without any input registered. Try to follow the video tutorial as well.

**Reply by bluebill1049** [reply]
here the video tutorial link: https://www.youtube.com/watch?v=-mFXqOaqgZk&feature=emb_logo follow it and let me know if this makes more sense afterwards. we can come back to your question again.

**Reply by kevinkli916** [reply]
Thanks for the material. I just read/watched through all of it but am not sure what you are pointing me towards. I am able to get validation to work, but not when I provide the registered inputs through a child component.

Currently, I have it structured as:
`<form>`
 `    <childComponent/>` <-- Registered inputs are in the child component
`</form>`

What am I missing for this to work?

**Reply by bluebill1049** [reply]
you should pass RHF methods into the child component, instead of creating a separate instance of hook form inside your child component.

**Reply by bluebill1049** [reply]
The following: 
```
const methods = useForm();
<ChildComponent {...methods} />
```

**Reply by kevinkli916** [reply]
Is there a way to do it without passing methods into the child component? I ask because in my case, the form wraps the routing that is rendering components. Each of the components have different inputs on them.

Ex:
```
<form>
    (mapping logic that renders components depending on conditions)
</form
```

**Reply by bluebill1049** [reply]
https://react-hook-form.com/api#useFormContext

**Reply by kevinkli916** [reply]
Thanks again for the quick replies. I'll take a look at this tomorrow and see if this will solve my problem.

**Reply by bluebill1049** [reply]
make sure you understand that you should only have one hook per form, not multiple hooks.

**Reply by kevinkli916** [reply]
Thanks for the help. FormContext did the trick for me.

**Comment by vairadurai007** [comment]
I have used the controller to the parent component to wrap the child component, the child component validation is not working, only initial validation is done. how to validate the form from parent to child component
